VisualRecognitionApi/src/main/java/javax/visrec/AbstractImageClassifier.java

```diff
@@ -4,21 +4,21 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.util.Map;
-import java.util.Optional;
-import javax.visrec.internal.ImageFactoryProvider;
 import javax.visrec.ml.classification.Classifier;
 import javax.visrec.util.Builder;
-import javax.visrec.util.ImageFactory;
 
 /**
  * Skeleton abstract class to make it easier to implement image classifier.
  * It provides implementation of Classifier interface for images, along with
  * image factory for specific type of images.
  *
+ * @author Zoran Sevarac <zoran.sevarac@deepnetts.com>
  * @param <IMAGE_CLASS>
  * @param <MODEL_CLASS>
- * @author Zoran Sevarac <zoran.sevarac@deepnetts.com>
+ *
+ * @deprecated Kevin: We should rethink the necessary use of this class in the API. I think we should move this one to the RI.
  */
+@Deprecated(forRemoval = true)
```

Comment:
Better remove than deprecate as our API is still under development.

Reply:
Yes, you're right. This is going to be moved to the RI.